src/coreclr/jit/emitarm64.cpp

```diff
@@ -11653,22 +11971,55 @@ void emitter::emitIns_R_R_R_R(instruction     ins,
             assert(isVectorRegister(reg1));
             assert(isPredicateRegister(reg2));
             assert(isGeneralRegister(reg3));
-            assert(isVectorRegister(reg4));
             assert(isScalableVectorSize(size));
             assert(insScalableOptsNone(sopt));
 
-            if (insOptsScalableDoubleWord32bitExtends(opt))
-            {
-                fmt = IF_SVE_HW_4A;
-            }
-            else if (insOptsScalableSingleWord32bitExtends(opt))
+            if (isGeneralRegisterOrZR(reg4))
             {
-                fmt = IF_SVE_HW_4A_A;
+                switch (ins)
+                {
+                    case INS_sve_ldff1b:
+                        assert(insOptsScalableStandard(opt));
+                        fmt = IF_SVE_IG_4A_E;
+                        break;
+
+                    case INS_sve_ldff1sb:
+                        assert(insOptsScalableAtLeastHalf(opt));
+                        fmt = IF_SVE_IG_4A_D;
+                        break;
+
+                    case INS_sve_ld1sb:
+                        assert(insOptsScalableAtLeastHalf(opt));
+                        fmt = IF_SVE_IK_4A_F;
+                        break;
+
+                    case INS_sve_ld1b:
+                        assert(insOptsScalableStandard(opt));
+                        fmt = IF_SVE_IK_4A_H;
+                        break;
+
+                    default:
+                        assert(!"Invalid instruction");
```

Comment:
this is little odd that we have 4 cases that can land us inside this switch/case and here too we need to add `default` and `assert`.

Reply:
The `default` case should never be hit here. The assert is more for sanity.

Reply:
I know, I am just pointing that we are writing odd looking pattern and testing C++ compiler here :) But honestly, not sure what's the best alternative here.

```c++
switch (ins)
{
  case insA:
  case insB:
  case insC:
  case insD:
      ...
      ...
      switch (ins)
      {
         case insA:
         case insB:
         case insC:
         case insD:
         ...
         default:
            assert("c++ compiler messed up :)");
      }
}
```